iati/core/tests/test_default.py

```diff
@@ -83,6 +84,42 @@ def test_default_ruleset(self):
 
         assert isinstance(ruleset, iati.core.Ruleset)
 
+    def test_default_ruleset_validation_rules_valid(self, schema_ruleset):
+        """Check that a fully valid IATI file does not raise any type of error (including rules/rulesets)."""
+        data = iati.core.tests.utilities.load_as_dataset('valid_std_ruleset')
+        result = iati.validator.full_validation(data, schema_ruleset)
+
+        assert iati.validator.is_xml(data.xml_str)
+        assert iati.validator.is_iati_xml(data, schema_ruleset)
+        assert not result.contains_errors()
+
+    @pytest.mark.parametrize("rule_error, invalid_dataset_name, help_text", [
+        ('err-rule-at-least-one-conformance-fail', 'invalid_std_ruleset_missing_sector_element', 'At least one of `sector` or `transaction/sector` must be present within each `//iati-activity`.'),
+        ('err-rule-date-order-conformance-fail', 'invalid_std_ruleset_bad_date_order', '`activity-date[@type=\'1\']/@iso-date` must be chronologically before `activity-date[@type=\'3\']/@iso-date` within each `//iati-activity`.'),
+        ('err-rule-regex-matches-conformance-fail', 'invalid_std_ruleset_bad_identifier', 'Each instance of `reporting-org/@ref` and `iati-identifier` and `participating-org/@ref` and `transaction/provider-org/@ref` and `transaction/receiver-org/@ref` within each `//iati-activity` must match the regular expression `[^\\/\\&\\|\\?]+`.'),
+        ('err-rule-sum-conformance-fail', 'invalid_std_ruleset_does_not_sum_100', 'Within each `//iati-activity`, the sum of values matched at `recipient-country/@percentage` and `recipient-region/@percentage` must be `100`.')
+        # Note the Rules relating to 'dependent', 'no_more_than_one', 'regex_no_matches', 'startswith' and 'unique' are not used in the Standard Ruleset.
+    ])
+    def test_default_ruleset_validation_rules_invalid(self, schema_ruleset, rule_error, invalid_dataset_name, help_text):
+        """Check that the expected rule error is detected when validating files containing invalid data for that rule.
+
+        Todo:
+            Consider whether this test should remove all warnings and assert that there is only the expected warning contained within the test file.
+
+            Check that the expected missing elements appear the the help text for the given element.
+        """
+        data = iati.core.tests.utilities.load_as_dataset(invalid_dataset_name)
+        result = iati.validator.full_validation(data, schema_ruleset)
+        errors_for_rule_error = result.get_errors_or_warnings_by_name(rule_error)
+        errors_for_ruleset = result.get_errors_or_warnings_by_name('err-ruleset-conformance-fail')
+
+        assert iati.validator.is_xml(data.xml_str)
+        assert iati.validator.is_iati_xml(data, schema_ruleset)
+        assert not iati.validator.is_valid(data, schema_ruleset)
+        assert len(errors_for_rule_error) == 1
+        assert len(errors_for_ruleset) == 1
+        assert help_text in errors_for_rule_error[0].help
```

Comment:
May be worth adding a note that this test could be a tad annoying to maintain should the string representation of Rules change.

`test_rule_string_output_general` and `test_rule_string_output_specific` in `test_rulesets.py` do something related for Rules.

...a more generic test to check that placeholder values have been replaced in relevant places would probably be useful.

Reply:
Added note in 59292385403362113a95eca65b6ab9b11591ff8c
Changed from `help` to `info` in 129fcc3aae2d74757223b952544ad8689420b6bd